django/db/backends/mysql/schema.py

```diff
@@ -65,6 +77,31 @@ def _field_should_be_indexed(self, model, field):
             return False
         return create_index
 
+    def _max_index_size(self):
+        """
+        Return the maximum index size, either from the user-provided
+        MAX_INDEX_CHARACTER_SIZE in the database settings, or try to guess the
+        number from the MySQL version and connection character set.
+        Generally ranges between 191 and 1024.
+        """
+        try:
+            return self.connection.settings_dict['MAX_INDEX_CHARACTER_SIZE']
+        except KeyError:
+            pass
+        size_in_bytes = 767 if self.connection.mysql_version < (5, 7, 7) else 3072
+        # *Guess* charset from connection charset
+        charset = self.connection.get_connection_params()['charset']
```

Comment:
Are we considering defaulting to using utf8mb4 for the connection? I wonder if it would make sense just to assume 191 if mysql_version < (5, 7, 7) and 767 if newer.

Reply:
Yeah, thinking about it more, I think either approach makes sense.